### PR TITLE
Use enc: dir

### PR DESCRIPTION
**Pull request by OR13**
This pull request simplifies some examples, adds support for HPKE Setup Info, and clarifies the use of "enc: dir" for both HPKE JWE modes.

### CODE REVIEW

File: draft-ietf-jose-hpke-encrypt.md [hunk 1]
```diff
@@ -207,12 +264,74 @@ In HPKE JWE Key Encryption:
 - The protected header MUST contain an "enc" that is registered in both the IANA HPKE AEAD Identifiers Registry, and the IANA JSON Web Signature and Encryption Algorithms Registry.
 - The recipient unprotected header parameters "psk_id" and "auth_kid" MAY be present.
 - The recipient unprotected header parameter "ek" MUST be present.
-- The recipient unprotected header parameter MAY contain "alg" and "enc" values.
+- The recipient unprotected header MUST contain an "alg" that starts with "HPKE".
```
Comment (selfissued) [inline draft-ietf-jose-hpke-encrypt.md:267]:
```suggestion
- The recipient unprotected header MUST contain a registered HPKE "alg" value.
```

File: draft-ietf-jose-hpke-encrypt.md [hunk 2]
```diff
@@ -183,19 +183,76 @@ Encapsulated keys MUST be the base64url encoded encapsulated key as defined in S
...
+~~~
+eyJhbGciOiJIUEtFLVAyNTYtU0hBMjU2LUExMjhHQ00iLCJlbmMiOiJkaXIiLCJraWQiOiJ1cm46aWV0ZjpwYXJhbXM6b2F1dGg6andrLXRodW1icHJpbnQ6c2hhLTI1Njp2b2RIQ3FjVVdFbV83NUpWcXlhTjhaS1FVMjF3VEFSYzhkRzhuVU1jZlBVIn0.BCsvYxTHM4CO_OwQxL3lkJDdlw3UDjx2xN9MIXnbVzfTgFJmo_Es2xdH-fYs9EXfH_V53JgMWfUm7rBD_oE5efU..7_va6cnwClMsw7h7lqpm2tCrH9NkciM-g9UabdPWcOeIRmAf01NLYG7Wn8fFoohHlcGgd0nh7Jmo9nvHFi7sH6kOX7pplBnvLUoPrqeyW4TdXo_X8YThNKf9BFyWGyF6fjelbic5jSYClFaenMkTnjpHxFW1sWuiuZVmO1EOzrlNttWy.
+~~~
+
+After verification:
+
+~~~
+{
+  "protectedHeader": {
+    "alg": "HPKE-P256-SHA256-A128GCM",
```
Comment (selfissued) [inline draft-ietf-jose-hpke-encrypt.md:214]: We should deal with this in a different issue, but this is the longest "alg" value ever - by far.  They're meant to be short registered identifiers - not strings you can parse.

File: draft-ietf-jose-hpke-encrypt.md [hunk 3]
```diff
@@ -183,19 +183,76 @@ Encapsulated keys MUST be the base64url encoded encapsulated key as defined in S
...
-In HPKE JWE Key Encryption, each recipient JWE Encrypted Key is the encrypted content encryption key, and the encapsulated key is found in the recipient header.
+In HPKE JWE Key Encryption, each recipient JWE Encrypted Key is the encrypted content encryption key, and the encapsulated key (ek) is found in the recipient header.
 
 # Integrated Encryption
 
 In HPKE JWE Integrated Encryption:
 
 - The protected header MUST contain an "alg" that starts with "HPKE".
-- The protected header MUST contain an "enc" that is registered in both the IANA HPKE AEAD Identifiers Registry, and the IANA JSON Web Signature and Encryption Algorithms Registry.
+- The protected header MUST contain an "enc" and it must be set to the value "dir".
```
Comment (tireddy2) [inline draft-ietf-jose-hpke-encrypt.md:193]: You may want to replace "must" with "MUST".

File: draft-ietf-jose-hpke-encrypt.md [hunk 4]
```diff
@@ -183,19 +183,76 @@ Encapsulated keys MUST be the base64url encoded encapsulated key as defined in S
...
 - The protected header parameters "psk_id" and "auth_kid" MAY be present.
 - The protected header parameters "ek" MUST NOT be present.
 - The "encrypted_key" MUST be the base64url encoded encapsulated key as defined in Section 5.1.1 of {{RFC9180}}.
 - The "iv", "tag" and "aad" members MUST NOT be present.
 - The "ciphertext" MUST be the base64url encoded ciphertext as defined in Section 5.2 of {{RFC9180}}.
+- The HPKE Setup info parameter MAY be used and its values are not constrained by this specification, by default it is empty.
```
Comment (tireddy2) [inline draft-ietf-jose-hpke-encrypt.md:199]: we will have to remove the following line from the draft: 
The "Setup info" MUST NOT be used with either HPKE JWE Integrated Encryption and HPKE JWE Key Encryption.

File: draft-ietf-jose-hpke-encrypt.md [hunk 5]
```diff
@@ -207,12 +264,74 @@ In HPKE JWE Key Encryption:
 - The protected header MUST contain an "enc" that is registered in both the IANA HPKE AEAD Identifiers Registry, and the IANA JSON Web Signature and Encryption Algorithms Registry.
 - The recipient unprotected header parameters "psk_id" and "auth_kid" MAY be present.
 - The recipient unprotected header parameter "ek" MUST be present.
-- The recipient unprotected header parameter MAY contain "alg" and "enc" values.
+- The recipient unprotected header MUST contain a registered HPKE "alg" value.
+- The recipient unprotected header MUST contain an "enc" and it must be set to the value "dir".
```
Comment (tireddy2) [inline draft-ietf-jose-hpke-encrypt.md:268]: In the case of Key Encryption, what is the need to convey "enc" in the recipient unprotected header? I don’t see a need for it in this mode, as the HPKE JWE Key Encryption Mode can function similarly to Key Agreement with ECDH-ES.

File: draft-ietf-jose-hpke-encrypt.md (hunk 4 above)
Comment (tireddy2) [reply draft-ietf-jose-hpke-encrypt.md:199]: I think for interoperability we should say the following for both modes :
When apu / apv are present, the HPKE Setup info parameter will carry the JOSE context-specific data defined in Section 4.6.2 of [RFC7518].

**Comment by OR13** [comment]
@tireddy2 would you mind merging this, and then making adjustments in https://github.com/ietf-wg-jose/draft-ietf-jose-hpke-encrypt/pull/9

We still have some further alignment to cover, but it would be good to have no open PRs during the week of IETF.

And we can consider publishing a new version when the draft window opens up.

**Review by selfissued — APPROVED** [review]

**Comment by selfissued** [comment]
Agreed - the spec will be improved by merging this.  I just approved.

**Comment by tireddy2** [comment]
@OR13 @selfissued - Merged the PR, please review https://github.com/ietf-wg-jose/draft-ietf-jose-hpke-encrypt/pull/9

**Pull request merged**